index.js

```diff
@@ -1,8 +1,8 @@
-module.exports = function(content) {
-  this.cacheable && this.cacheable();
-  this.value = content;
+module.exports = function (content) {
+  this.cacheable && this.cacheable()
+  this.value = content
 
-  return 'module.exports = Buffer.from("' + content.toString('base64') + '", "base64")';
-};
+  return `module.exports = Buffer.from("${content}")`
```

Comment:
I believe content needs to be encoded as being binary data, it may contain characters which would break the naïve approach here, e.g. `"`, `\n`, `\0`, etc.

Reply:
The Webpack docs seem to be wrong, what Webpack passes your loader is actually a `Buffer` object: this is why we can call `.toString('base64')` on it, for example, which wouldn't work with a string. In general, it wouldn't make sense to load an arbitrary file as a string so it has to work this way.

When you write `${content}` though, you are asking this `Buffer` to convert itself to a string. `Buffer` is smart enough to escape control characters to escape sequences (e.g. `\u0000`) but as @nhardy says, it doesn't know that you are pasting that string between double quotes so it will not escape `"`, `\` or `\n`, breaking the generated source. Also, `Buffer` will try to interpret the file's contents as UTF-8 and make a mess of things if the file is anything else.

What exactly is the problem you are encountering with the test?

Reply:
Aah, the trick is that because you have `module.exports.raw = true;` in the loader, Webpack calls it with a `Buffer` and not a string. So the docs are probably accurate for the general case, but I can't find documentation for that `raw` flag.

Reply:
Ahh, makes sense. Thank you! 
I've pushed a new commit, please have a look again 🙏